### PR TITLE
Fix #1182: pickers use entry.filename as default and fallback to entry.value

### LINKED ISSUE #1182: rg: file_ignore_patterns will be ignored in searches
### Description

This is my setup:
```
local telescope = require("telescope")
telescope.setup {
    defaults = {
        file_sorter = require "telescope.sorters".get_fzy_sorter,
        generic_sorter = require "telescope.sorters".get_fzy_sorter,
        vimgrep_arguments = {
            "rg",
            "--color=never",
            "--no-heading",
            "--with-filename",
            "--line-number",
            "--column",
            "--smart-case",
            "--hidden"
        },
        file_ignore_patterns = {
            "node_modules",
            ".work/.*",
            ".cache/.*",
            ".idea/.*",
            "dist/.*",
            ".git/.*"
        },
       ... some other stuff ...
    }
}
```

When I use `live_grep` and search for `work.build` nothing is found. I have a lot of files including the string `.work/build/...`. When I remove the pattern `".work/.*"` from `file_ignore_patterns` I find all the matches and the output matches that of `rg` run in a shell.

Perhaps I am doing something wrong here.

### Neovim version

NVIM v0.6.0-dev+nightly-353-gff0833cb4

### Operating system and version

macOS 11.5

### Steps to reproduce

See above.

### Expected behavior

`file_ignore_patterns` should only affect the files that are searched.

### Actual behavior

See above.

### Minimal config

```Lua
See above.
```

### PR DESCRIPTION
**Pull request by TheLeoP**
Hi again @fdschmidt93 .

So, the diff is just two lines and it's pretty much exactly what you suggest on #1182 issue. I'm ready to make any changes if needed.

### CODE REVIEW

File: lua/telescope/pickers.lua [hunk 1]
```diff
@@ -1,6 +1,7 @@
 require "telescope"
 
 local a = vim.api
+local if_nil = vim.F.if_nil
```
Comment (fdschmidt93) [inline lua/telescope/pickers.lua:4]: Small nit, just leave `vim.F.if_nil` for now. We'll migrate from `utils.get_default` to internal functions eventually. Therefore better not to introduce shorthands for now here.

**Comment by fdschmidt93** [comment]
Did you test that this works well with the `builtin.{live_grep, grep_string, find_files}` pickers as I'd hoped? I'll have time to test drive this latest on Saturday myself.

**Review by TheLeoP — COMMENTED** [review]
Undesrtood

**Comment by TheLeoP** [comment]
Yes, I tested the change for `builtin.{live_grep, grep_string, find_files}` and it worked as expected (or at least it seemed to work as expected).

**Comment by TheLeoP** [comment]
I correct myself: the change work as expected on `builtin.{live_grep, grep_string}`. But, not mattering the config, find_files always shows files like `work/should_not_show` and never shows files like `.work/should_not_show`

**Comment by Conni2461** [comment]
whats the actual config? i briefly tested it and it works for me

**Comment by TheLeoP** [comment]
This is the config:
```lua
telescope.setup {
    defaults = {
        vimgrep_arguments = {
            "rg",
            "--color=never",
            "--no-heading",
            "--with-filename",
            "--line-number",
            "--column",
            "--smart-case",
            "--hidden"
        },
        file_ignore_patterns = {
            ".work\\.*",
            ".work/.*",
        },
    }
}
```

I guess it's because of the program I'm using with find_files and the fact that I'm not passing any arguments to it (so it doesn't show the files starting with a dot because they are hidden). If that's the case, the change should be working as intended.

**Comment by Conni2461** [comment]
It works for me i just tested it. Thanks for the fix.

The file ignore pattern is not a 100% correct lua regex. You should read `:help telescope.defaults.file_ignore_patterns`.

If you want to ignore all files under the directory work. Then it would be `work/.*` if you only want to ignore all files under the directory .work it would be `%.work/.*`. etc

**Pull request merged**